crates/sui-graphql-rpc/src/types/suins_registration.rs

```diff
@@ -57,6 +65,15 @@ pub(crate) struct SuinsRegistration {
     pub native: NativeSuinsRegistration,
 }
 
+/// Represents the results of a query for a domain's `NameRecord` and its parent's `NameRecord`. The
+/// `expiration_timestamp_ms` on the name records are compared to the checkpoint's timestamp to
+/// check that the domain is not expired.
+pub(crate) struct DomainExpiration {
+    pub parent_name_record: Option<NameRecord>,
+    pub name_record: Option<NameRecord>,
+    pub checkpoint_timestamp_ms: u64,
```

Comment:
nit: Can you add some doc comments to these fields? In particular what timestamp does `checkpoint_timestamp_ms` represent? I'm assuming that it's the timestamp for the checkpoint being viewed at, but it would be good to make that clear.